simple_salesforce/api.py

```diff
@@ -202,6 +205,27 @@ def restful(self,path,params):
             return None
         else:
             return json_result
+    
+    # Generic Tooling API Function
+    def tooling(self,path,params):
+        """Allows you to make a direct Tooling REST call if you know the path
+
+        Arguments:
+
+        * path: The path of the request
+            Example: sobjects/User'
+        * params: dict of parameters to pass to the path
+        """
+
+        url = self.tool_url + path
+        result = self.request.get(url, headers=self.headers, params=params)
+        if result.status_code != 200:
+            raise SalesforceGeneralError(result.content)
```

Comment:
Can you do `_exception_handler(result)` (see bottom of `api.py`) instead of throwing `SalesforceGeneralError()` directly?

I now see that there are other parts of the file that don't use that (and I'll have to fix them) but it's the only way for folks to correctly capture the correct exception for the result.

Thanks